### PR TITLE
refactor(tracing): change dir

### PR DESCRIPTION
**Pull request by yejiayu**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #

**Special notes for your reviewer**:

/cc @ddysher @kdada 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ddysher** [comment]
@yejiayu test is failing

File: plugins/tracing/client/http.go [hunk 1]
```diff
@@ -14,12 +14,13 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package tracing
+package client
```
Comment (ddysher) [inline plugins/tracing/client/http.go]: will this become a problem when we add more clients?  e.g. name conflict, client.Transport is required for both http and other protocol.

/cc @kdada

Comment (yejiayu) [reply plugins/tracing/client/http.go]: We can use the protocol name as a prefix

Comment (kdada) [reply plugins/tracing/client/http.go]: `plugins/tracing/clients/http/client.go` is more flexible.

Comment (ddysher) [reply plugins/tracing/client/http.go]: > plugins/tracing/clients/http/client.go is more flexible.

SGTM

**Comment by yejiayu** [comment]
ping @ddysher

**Pull request merged**